src/core/middlewares/chain.go

```diff
@@ -59,6 +61,7 @@ func (b *DefaultCreator) Create() *alice.Chain {
 
 func (b *DefaultCreator) geMiddleware(mName string) alice.Constructor {
 	middlewares := map[string]alice.Constructor{
+		CHART:            func(next http.Handler) http.Handler { return chart.New(next) },
```

Comment:
Can we create a separate chain for chart as the requests sent to registry don't need to be handled by the chart handler?

Reply:
`DefaultCreator` is the factory to make chain, it can create chain with different middleware and it is not the creator for registry only.